contracts/token/ERC721_base.cairo

```diff
@@ -447,26 +447,24 @@ func _check_onERC721Received{
         data: felt*
     ) -> (success: felt):
     let (caller) = get_caller_address()
-    # IAccount_ID = 0x50b70dcb
-    let (is_account) = IERC165.supportsInterface(to, 0x50b70dcb) 
-    if is_account == 1:
+    # ERC721_RECEIVER_ID = 0x150b7a02
+    let (is_supported) = IERC165.supportsInterface(to, 0x150b7a02)
+    if is_supported == 1:
+        let (selector) = IERC721_Receiver.onERC721Received(
+            to, 
+            caller, 
+            _from, 
+            token_id, 
+            data_len, 
+            data
+        )
+
+        # ERC721_RECEIVER_ID
+        assert (selector) = 0x150b7a02
```

Comment:
are the parenthesis needed? if so, then we could remove them from the previous statement to avoid deconstructing the return value